modules/platforms/dotnet/Apache.Ignite.Core/Client/Services/IServicesClient.cs

```diff
@@ -51,5 +51,12 @@ public interface IServicesClient
         /// </summary>
         /// <returns>Instance with server-side binary mode enabled.</returns>
         IServicesClient WithServerKeepBinary();
+
+        /// <summary>
+        /// Returns an instance for obtaining .Net services.
+        /// Service method serialization and deserialization will be .Net specific.
+        /// </summary>
+        /// <returns>Instance for obtaining platform services.</returns>
+        IServicesClient ForDotNetService();
```

Comment:
I think we should use the same approach as in thick service API and detect service type automatically. Let's not shift this responsibility to the user.

https://github.com/apache/ignite/blob/cca909ca92965aa12dca5139ec2803df4ffbc4d4/modules/platforms/dotnet/Apache.Ignite.Core/Impl/Services/Services.cs#L392

Reply:
We may want to implement https://issues.apache.org/jira/browse/IGNITE-13359 first for that.